### DISCUSSION TITLE
Changing a cc value

### DISCUSSION THREAD

**Discussion by HNT1807** [orchetect/swift-midi#214] [Q&A]
Hello and thank you so much for your help!

(For a macos app) I don't understand how to change a cc value of a midi file, for example, I want the user to specify what cc needs to be changed and to what value, let's say the user wants to change the velocity of all the notes to 100, how to do that?

**Comment by orchetect** [comment]
1. Iterate on the `MIDIFile`'s track chunks
2. For each track iterate on its events by index
3. Check the event type by using switch case and if it's a matching event (`noteOn`, `noteOff`, `cc`, etc.) then modify its velocity (notes) or value (cc).
